hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -195,6 +193,11 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		}
 	}
 
+	platformHandler, err := r.platformProviders.GetPlatformReconciler(hcluster.Spec.Platform.Type)
+	if err != nil {
+		return ctrl.Result{}, err
```

Comment:
Should probably set a new condition here or consider one of the existing ones with a reason such as "invalid a platform" or similar.